lib/Sema/TypeCheckConcurrency.cpp

```diff
@@ -2116,21 +2116,25 @@ namespace {
 
         auto behavior = DiagnosticBehavior::Error;
 
+        // Limit behavior if @preconcurrency detected
+        if (errors.front().preconcurrency)
```

Comment:
We can't just look at the first error here. We'll need to check that all of the errors are `.preconcurrency` before we can downgrade the error to a warning.